.github/workflows/measure_length_scales.yml

```diff
@@ -0,0 +1,53 @@
+name: CI
+
+on:
+  pull_request:
+  push:
+    branches:
+      - main
+  schedule:
+      - cron: "0 13 * * 1"  # Every Monday at 9AM EST
+
+jobs:
+  measure-length-scales:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v4
+
+      - name: Setup environment
+        run: |
+            python -m pip install --upgrade pip
+            pip install numpy
+            pip install imageruler
+            pip install ceviche-challenges
+
+      - name: cavity_design
+        run: |
+          cd cavity_design
+          python minimum_lengthscales_run.py
```

Comment:
Do these scripts check that the lengthscales haven't changed?

Reply:
No, this job just checks that the scripts can run. The scripts themselves never checked against expected values, they just measure the length scale.

Reply:
As long as this PR is adding test scripts, shouldn't it check the output?  At least a "golden master" test to alert us if the values change?

Reply:
This PR is not adding any new scripts, it is just updating existing scripts so that they can actually run (driven by imageruler changes), and adding a GitHub workflow that ensures the scripts run successfully.

If we want to compare against golden values, I think the better place to do this is the imageruler repo, so that it can help catch issues with any changes to the imageruler code. I think this would be a good addition.

We could do it in this repo as well, but it would probably make sense to do this with dedicated tests rather than the existing scripts. For one, at least one existing script is quite heavyweight (the mode converter, which runs a simulation to evaluate device performance), and running this for the dozens of designs in the repo would be unnecessarily slow.

Adding a test should be straightforward, but I suggest we create an issue and handle it in a follow up pr.